designs/authentication-use-cases.md

```diff
@@ -0,0 +1,192 @@
+
+The process of supplying extra or supplementary authentication to the authenticator. This improves
+the authentication from "there is a person present" to "this specific owner of the authenticator" is
+present. Conseder UV=True as the authenticator is a self container MFA device, where UV=false means that
+the device is a SFA, and other authentication is required to compose a complete MFA.
+
+At registration we store the policy and verification that was supplied in the internal Credential
+type so that later decisions can be made for security policy.
+
+### External Knowledege
+
+This is where using elements from the user interaction, workflow, or a device fingerprint, cookie
+or other, the client or RP can make descions about which credential or policy to use before
+the initial webauthn challenge is sent.
+
+This work flow could be a user login page that requests "how" the user wishes to authenticate,
+which device they want to use. It could also be based on the device ID and understanding that
+the authenticator is part of the device, so we should prefer it's use.
+
+## Authentication Work Flows
+
+### Homogenous Non-Discoverable Credentials
+
+This is a set of credentials with a homogeneous UV policy - either all discouraged or all required.
+
+This policy is derived from the set of UV=true/false flags in the allow credential list. If the
+UV flags are inconsistent an error is raised.
+
+In the case all of these credentials are discouraged, we still assert that the UV bit matches
+the returned authentication as it *is* valid for UV=true even when discouraged and we store that
+in the credential.
+
+#### Example
+
+A laptop where the user has multiple UV=false yubikeys that may be used as a authentication factor.
+
+A user indicates they want to authenticate with their touchid as a self contained MFA. The user owns
+and has enrolled multiple devices that have TouchID so any of them could be the credential in use
+without external knowledge.
+
+#### Detailed Example
+
+This has many interactions with the UV policy at registration, and what the resulting states and outcomes
+are.
+
+| UV Policy | UV boolean returned | Valid UV Policies for Auth | UV boolean always checked in auth |
+| --------- | ------------------- | -------------------------- | --------------------------------- |
+| discouraged | false             | discouraged                | No - device can not do UV |
+| discouraged | true              | discouraged, required      | Yes - device always does UV |
+| preferred   | false             | discouraged                | No - device can not do UV |
+| preferred   | true              | discouraged, required      | No - device may not always send UV in discouraged mode |
+| required    | false             | -                          | No - device can not be used |
+| required    | true              | required                   | Yes - device should always performs UV |
+
```

Comment:
It may also be worth making clear why preferred isn't a valid policy for auth (credentials registered with discouraged can be forced to use UV=true even though it wasn't true at registration, breaking user expectations, and we don't have the internal knowledge to know it's possible )